client-java/src/main/java/uk/gov/gchq/palisade/client/internal/dft/DefaultQueryResponse.java

```diff
@@ -0,0 +1,237 @@
+
+import java.util.Optional;
+import java.util.concurrent.Flow.Publisher;
+import java.util.concurrent.TimeUnit;
+import java.util.function.UnaryOperator;
+
+/**
+ * Default implementation for the "dft" subname
+ *
+ * @since 0.5.0
+ */
+public class DefaultQueryResponse implements QueryResponse {
+
+    /**
+     * Enum to reflect the state of reading from the websocket.
+     *
+     * @since 0.5.0
+     */
+    private enum Loop {
+        // normal state, keep reading websocket buffer
+        CONTINUE,
+        // No more left and we are done
+        COMPLETE,
+        // someone cancelled, so we should get out
+        CANCELLED
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultQueryResponse.class);
+
+    private final DefaultSession session;
+    private final PalisadeResponse palisadeResponse;
+
+    /**
+     * Returns a newly created {@code DefaultQueryResponse} with the provided
+     * {@code session} and {@code palisadeResponse}
+     *
+     * @param session          The session providing connection to the cluster
+     * @param palisadeResponse The response from the cluster
+     */
+    public DefaultQueryResponse(final DefaultSession session, final PalisadeResponse palisadeResponse) {
+        this.session = session;
+        this.palisadeResponse = palisadeResponse;
+    }
+
+    @SuppressWarnings("java:S3776")
+    @Override
+    public Publisher<Message> stream() {
+
+        // our flowable must wrap the websocket client
+
+        var flowable = Flowable.create((final FlowableEmitter<Message> emitter) -> {
+
+            LOGGER.debug("Creating stream...");
+
+            var configuration = session.getConfiguration();
+            var webSocketClient = WebSocketClient.createResourceClient(b -> b
+                .token(palisadeResponse.getToken())
+                .objectMapper(configuration.getObjectMapper())
+                .baseUri(configuration.getFilteredResourceUrl()));
+
+            webSocketClient.connect();
+
+            LOGGER.debug("Connected to websocket");
+
+            var loop = Loop.CONTINUE;
+            do {
+                var wsm = webSocketClient.poll(1, TimeUnit.SECONDS);
+                if (wsm != null) {
+                    if (wsm instanceof CompleteMessage) {
+                        loop = Loop.COMPLETE;
+                    } else {
+                        createMessage(wsm).ifPresent((final Message msg) -> {
+                            LOGGER.debug("emitter.onNext: {}", msg);
+                            emitter.onNext(msg);
+                        });
+                    }
+                }
+                if (emitter.isCancelled()) {
+                    loop = Loop.CANCELLED;
+                }
+            } while (loop == Loop.CONTINUE);
+
+            if (loop == Loop.COMPLETE) {
+                LOGGER.debug("emitter.complete");
+                emitter.onComplete();
+            } else if (loop == Loop.CANCELLED) {
```

Comment:
loop is always Loop.CANCELLED

Reply:
During testing i get 
```
08:43:11.203 DEBUG u.g.g.p.c.i.dft.DefaultQueryResponse - emitter.complete
```
in the log. So it does hit line 120.

Reply:
The inspection here is `loop !!= Loop.CONTINUE` because of the `while loop == Loop.CONTINUE`
So:
```
            if (loop == Loop.COMPLETE) {
                LOGGER.debug("emitter.complete");
                emitter.onComplete();
            } else {
                // else Loop.CANCELLED
                LOGGER.debug("emitter.cancelled");
            } // cannot ever be Loop.CONTINUE
```

Reply:
I've added a log message to the final else. It's now:
```
if (loop == Loop.COMPLETE) {
    LOGGER.debug("emitter.complete");
    emitter.onComplete();
} else if (loop == Loop.CANCELLED) {
    LOGGER.debug("emitter.cancelled");
} else {
    LOGGER.debug("emitter ended normally");
}
```
I think this makes more sense.

Reply:
The extra else was not needed. I bit cleaner now.

Reply:
Done.